source/adios2/toolkit/sst/dp/daos_dp.c

```diff
@@ -620,7 +620,7 @@ static void *DaosReadRemoteMemory(CP_Services Svcs, DP_RS_Stream Stream_v,
     char *StringName = malloc(20);
     void *NVBlock;
     char *BaseAddr;
-    sprintf(StringName, "Timestep_%ld", Timestep);
+    snprintf(StringName, 20, "Timestep_%ld", Timestep);
     BaseAddr =
         NULL; // nvs_get_with_malloc(Stream->writer_nvs[Rank], StringName, 1);
 
```

Comment:
```
Line 701:    //    sprintf(Info->CheckString, "Daos info for timestep %ld from rank
```
I know this is a comment but if someone decided we will need this logic for DAOS and removes the comment we will get the warning again

And the same in `source/adios2/toolkit/sst/dp/evpath_dp.c`:
```
//    sprintf(Info->CheckString, "Evpath info for timestep %ld from rank
```

Reply:
Well, I saw the HDF5 uses, but they are the messier ones.  For example:

```
size_t gGetNameOfNthItem(H5VL_ObjDef_t *vol, uint32_t idx, char *name)
{
 . . . 
    if (NULL != name)
    {
        sprintf(name, "%s", n);
    }
```
Here, the target of the sprintf() is passed in to the subroutine without length information, so it's a non-local change.  (Then again, why this particular use is sprintf() rather than strcat() isn't immediately obvious.)  So, I decided not to touch HDF5 here.  (I also don't run with hdf5 on my mac, so I'm not seeing those warnings either.  Maybe we can upgrade the XCode in our Mac CI at some point and we'll tackle them there.  Either that or @guj can tackle it?)

WRT code in comments, I'm always a torn about including it when making these sorts of changes.  Some folks would argue that code shouldn't be commented out to begin with, because old code is always available in version control if you want to dig for it. But given that code exists that is commented out or surrounded by untriggered #ifdef's, I think changing it in a PR in the hopes that everything will be fine if it ever gets uncommented is optimistic.  Even if it used to compile the last time it was active, anything could have rendered it invalid, including errors in this sort of change and nobody would know because it's a comment.  Compiling with a non-fatal warning is probably one of the better outcomes one could hope for, so I left those cases alone...  Happy to reconsider if someone feels strongly though.

Reply:
I am in favor of removing all commented code, but I see your point. The PR looks good to me.

There are actually several things in the HDF5 file that I would change. If we ever need to run with this backend it might be worthwhile to look at this code.

Reply:
About the comments,     maybe add reason why they  are commented out along with commented code if not removing them? 

Greg (@eisenhauer) I can make the equivalent  changes for HDF5 related files  in a separate pull request.

Ana (@anagainaru) do you want to list what are the other things  you would change. for HDF5 files?

Reply:
@guj only one comment that I actually care about. We define in the `H5VolError.h` heade file:
```c++
void *safe_calloc(size_t n, size_t s, unsigned long line);
#define SAFE_CALLOC(n, s) safe_calloc(n, s, __LINE__)
```
not in a namespace but as global names (function/macro). If a code includes this and reuses the same name it will get errors. Or if HDF5 decides to create them, since we do `#include <hdf5.h>` within this code, ADIOS2 will not compile anymore. I would prefix all of them with `ADIOS2_`

The rest of the comments are more coding style related. I like decreasing the amount of repeated code but it's not essential. There's also some commented code that I doubt you will ever need again. Nothing too critical.

Reply:
> 
>   I would prefix all of them with `ADIOS2_`
> 
Good point. Will include it in next PR.  Thanks @anagainaru